rust/stub/src/main.rs

```diff
@@ -162,40 +169,26 @@ fn main(handle: Handle, mut system_table: SystemTable<Boot>) -> Status {
         return Status::SECURITY_VIOLATION;
     }
 
-    let kernel_cmdline =
-        booted_image_cmdline(system_table.boot_services()).expect("Failed to fetch command line");
-
-    let kernel_handle = {
-        system_table
-            .boot_services()
-            .load_image(
-                handle,
-                uefi::table::boot::LoadImageSource::FromBuffer {
-                    buffer: &kernel_data,
-                    file_path: None,
-                },
-            )
-            .expect("UEFI refused to load the kernel image. It may not be signed or it may not have an EFI stub.")
-    };
+    let kernel = unsafe { pe_loader::load(system_table.boot_services(), &kernel_data) }
+        .expect("Failed to load the kernel");
 
-    let mut kernel_image = system_table
+    let mut loaded_image = system_table
         .boot_services()
-        .open_protocol_exclusive::<LoadedImage>(kernel_handle)
+        .open_protocol_exclusive::<LoadedImage>(handle)
```

Comment:
Does anyone know how to create a new UEFI handle? It seems wrong to re-use the lanzaboote handle for the kernel we load. Or is this the way to go?

@RaitoBezarius

Reply:
We can locate a (set of) handles which can be opened for the LoadedImage protocol using `locate_protocol`: https://docs.rs/uefi/0.1.2/uefi/struct.BootServices.html#method.locate_handle_by_protocol AFAIK

But, it's going to probably return one of the handle we are holding right now.

Reply:
I don't see a way to allocate a new UEFI handle. LocateProtocol is definitely a bad choice: that returns "the first handle" supporting the protocol, which is some driver or systemd-boot, making it a much worse choice than our own handle. At least the uefi crate documents using `set_image` for pretty much our exact use-case, and shim also does the same.

Reply:
I do think this method returns the set of *all* handles supporting the protocol, but you're right.

Reply:
Thanks for looking into this.